client/src/main/scala/org/http4s/client/DefaultClient.scala

```diff
@@ -230,6 +230,7 @@ private[http4s] abstract class DefaultClient[F[_]](implicit F: Bracket[F, Throwa
   def get[A](s: String)(f: Response[F] => F[A]): F[A] =
     Uri.fromString(s).fold(F.raiseError, uri => get(uri)(f))
 
-  private def defaultOnError(resp: Response[F])(implicit F: Applicative[F]): F[Throwable] =
-    F.pure(UnexpectedStatus(resp.status))
+  private def defaultOnError(req: Request[F], resp: Response[F])(
```

Comment:
What if you curried this:

```suggestion
  private def defaultOnError(req: Request[F])(resp: Response[F])(
```

Then the `expect`s look like:

```scala
expectOr(req)(defaultOnError(req))
```